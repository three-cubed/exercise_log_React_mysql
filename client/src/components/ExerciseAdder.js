import React, { useState } from 'react';

const ExerciseAdder = ({ addExerciseEvent }) => {
    const [exerciseEventTitle, setExerciseEventTitle] = useState('');
    const [exerciseDate, setExerciseDate] = useState('');
    const [achievement, setAchievement] = useState('');
    const [colour, setColour] = useState('');

    const submitExerciseEvent = (e) => {
        e.preventDefault();
        if (!exerciseEventTitle || !exerciseDate) {
            alert('Make sure you have added an exercise event and when you did it!');
            return;
        }
        addExerciseEvent({ exerciseEventTitle, exerciseDate, achievement, colour });
        setExerciseEventTitle('');
        setExerciseDate('');
        setAchievement('');
        // colour not to be reset, can be left as it is (which is as it is displayed) 
    }

    return (
        <div className='formContainer'>
            <form id='exerciseAdderForm' 
                data-testid='exerciseAdderForm' 
                onSubmit={submitExerciseEvent}
            >
                <div>
                    <br /><label>Exercise:</label><br />
                    <input type='text' placeholder='What was the exercise?' aria-label="titleInput" 
                    value={exerciseEventTitle} 
                    onChange={(e) => setExerciseEventTitle(e.target.value)}
                    />
                </div>
                <div>
                    <br /><label>Day:</label><br />
                    <input type='text' placeholder='When did you do it?' aria-label="dateInput" 
                    value={exerciseDate} 
                    onChange={(e) => setExerciseDate(e.target.value)}
                    />
                </div>
                <div>
                    <br /><label>Achievement:</label><br />
                    <input type='text' placeholder='What did you manage to do?' 
                    value={achievement} 
                    onChange={(e) => setAchievement(e.target.value)}
                    />
                </div>
                <div>
                    <br /><label>Exercise colour:</label><br />
                    <select name="topic" onChange={(e) => setColour(e.target.value)}>
                            <option value="grey">Grey</option>
                            <option value="black">Black</option>
                            <option value="firebrick">Red</option>
                            <option value="coral">Orange</option>
                            <option value="gold">Yellow</option>
                            <option value="green">Green</option>
                            <option value="blue">Blue</option>
                            <option value="purple">Purple</option>
                            <option value="pink">Pink</option>
                            <option value="saddlebrown">Brown</option>
                    </select>
                </div>
                <br />
                <div className='centred'>
                    <button type='submit' id='submissionBtn' data-testid='submissionBtn'>
                        Save new exercise event!
                    </button>
                </div>
            </form>
        </div>
    )
}

export default ExerciseAdder;
