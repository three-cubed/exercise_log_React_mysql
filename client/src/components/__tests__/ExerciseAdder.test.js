import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ExerciseAdder from '../ExerciseAdder.js';

afterEach(cleanup);

describe('ExerciseAdder component', () => {
    test('renders', () => {
        render(<ExerciseAdder />);
    });
    test('matches snapshot', () => {
        const tree = renderer.create(<ExerciseAdder />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('pressing submissionBtn sets off submission of form', () => {
        const mockFunction = jest.fn();
        const dummySubmit = () => {
            window.alert = () => {};  // provides an empty implementation for window.alert
            mockFunction();
        }
        const renderExerciseAdder = render(<ExerciseAdder onSubmit={dummySubmit()} />);
        const submissionBtn = renderExerciseAdder.getByTestId('submissionBtn');
        fireEvent.click(submissionBtn);
        expect(mockFunction).toBeCalled();
    });
})
