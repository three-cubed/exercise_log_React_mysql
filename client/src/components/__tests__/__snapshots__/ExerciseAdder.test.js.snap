// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseAdder component matches snapshot 1`] = `
<div
  className="formContainer"
>
  <form
    data-testid="exerciseAdderForm"
    id="exerciseAdderForm"
    onSubmit={[Function]}
  >
    <div>
      <br />
      <label>
        Exercise:
      </label>
      <br />
      <input
        aria-label="titleInput"
        onChange={[Function]}
        placeholder="What was the exercise?"
        type="text"
        value=""
      />
    </div>
    <div>
      <br />
      <label>
        Day:
      </label>
      <br />
      <input
        aria-label="dateInput"
        onChange={[Function]}
        placeholder="When did you do it?"
        type="text"
        value=""
      />
    </div>
    <div>
      <br />
      <label>
        Achievement:
      </label>
      <br />
      <input
        onChange={[Function]}
        placeholder="What did you manage to do?"
        type="text"
        value=""
      />
    </div>
    <div>
      <br />
      <label>
        Exercise colour:
      </label>
      <br />
      <select
        name="topic"
        onChange={[Function]}
      >
        <option
          value="grey"
        >
          Grey
        </option>
        <option
          value="black"
        >
          Black
        </option>
        <option
          value="firebrick"
        >
          Red
        </option>
        <option
          value="coral"
        >
          Orange
        </option>
        <option
          value="gold"
        >
          Yellow
        </option>
        <option
          value="green"
        >
          Green
        </option>
        <option
          value="blue"
        >
          Blue
        </option>
        <option
          value="purple"
        >
          Purple
        </option>
        <option
          value="pink"
        >
          Pink
        </option>
        <option
          value="saddlebrown"
        >
          Brown
        </option>
      </select>
    </div>
    <br />
    <div
      className="centred"
    >
      <button
        data-testid="submissionBtn"
        id="submissionBtn"
        type="submit"
      >
        Save new exercise event!
      </button>
    </div>
  </form>
</div>
`;
